### ISSUE TITLE
KB3173424 requires KB2919355

### ISSUE THREAD

**Issue by olivierlemasle** [criteo-cookbooks/ms_dotnet#45]
On my Windows 2012R2 box (with no update), I cannot install KB3173424 (marked in the cookbook as a MS .Net 4.6 requirement) because it requires KB2919355.

This dependency is documented here:
https://support.microsoft.com/en-us/help/3173424/servicing-stack-update-for-windows-8.1-and-windows-server-2012-r2-july-12,-2016

If I revert commit e20a0ed4b, everything works in my case.
But looking at this repository's history, we can see that it's not that simple:

- .Net 4.6 depends on KB2919355, so this update has been added in #25 (to fix #24)
- Then, as KB2919355 depends on KB2919442, the "prerequisites of the prerequisites" were added in #27.
- Then KB2919442 was replaced by KB3021910 in #31 to fix #29 (and for me it works perfectly)
- Then KB3021910 was replaced by KB3173424 in #38 to fix #37 
... and KB3173424 depends on KB2919355, which is a nice loop!

So I'd be happy to revert to KB3021910 (which did not depend on anything), but it would cause #37 again (found by both @lbn-chef-team and @Jakauppila).

Any idea?

**Comment by Annih** [comment]
Interesting, in that kind of situation I love Microsoft :)
I think one possible solution would be to install all updates, even the one that have been superseeded by newer updates.

On my 2012R2 box I nver encountered your issue because KB2919355 is already included.

I'm testing it right away!

**Comment by Annih** [comment]
Having all packages (KB2919442 KB2919355 KB3021910 KB3173424) as prerequisites seem to fix the issue.
I'm trying to remove KB3021910, which doesn't seem useful anymore

**Comment by olivierlemasle** [comment]
Thanks! In which order do you propose to install these updates?

Why won't this change re-open issue #29? It seems that the users reporting this issue were not able to install KB2919442 at the first place, probably because they already had other updates already installed.

**Comment by Annih** [comment]
Ok so not installing KB3021910 seems not an issue :)
All my test have been done on a Windows Server 2012R2 box with KB2919442 included by default and not removable ...

I'm now trying to reproduce on a box without this KB

**Comment by olivierlemasle** [comment]
Thanks, I'm testing on my side KB2919442 + KB2919355 + KB3173424

**Comment by olivierlemasle** [comment]
I confirm that it works for me with these three updates. I've added a pull request.

**Comment by Annih** [comment]
Thanks for the feedback and PR :)

**Comment by Annih** [comment]
Following #50 I decided to reopen this issue to continue the discussion until all usecase seems fixed :)

1. It seems that it is possible to have KB3173424 without KB2919355 even if it is one of its prerequisites...
2. Having KB3173424 is not enough to install .NET you still need KB2919355...
3. KB2919355 dépends on KB2919442 (#46)
4. KB2919442 can't be installed when KB3173424 ...

I created #51 to test a new solution. I think I'll list all updates superseeding KB2919442
